enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -367,6 +367,9 @@ func (h *executorHandle) Serve(ctx context.Context) error {
 			} else if req.GetAskForMoreWorkRequest() != nil {
 				node := h.getRegistration()
 				poolKey := nodePoolKey{os: node.GetOs(), arch: node.GetArch(), pool: node.GetPool()}
+				if h.scheduler.enableUserOwnedExecutors {
+					poolKey.groupID = h.GroupID()
+				}
```

Comment:
could we add a method on the scheduler object that returns the poolKey and use it here and elsewhere we have this same code?

Reply:
done